module/VuFind/src/VuFind/Controller/MyResearchController.php

```diff
@@ -1220,6 +1253,155 @@ public function holdsAction()
         return $view;
     }
 
+    /**
+     * Edit holds
+     *
+     * @return mixed
+     */
+    public function editHoldsAction()
+    {
```

Comment:
I tried initially to put this in the holds plugin like `cancelHolds` is done, but interfacing it with the action method became too complicated. I suppose something like separate controllers for different parts of MyResearch (like LibraryCardsController now) would be a more natural way of splitting things up.

Reply:
Yes, I have been wanting to split up MyResearchController for years, but I just never find the time since it's not broken, it's just ugly. Ideally, at the same time we split this up, we could also switch to dependency-injected controllers to move away from explicit reliance on the service manager (as discussed in [VUFIND-966](https://vufind.org/jira/browse/VUFIND-966)). So many improvement projects, so little time! Unfortunately, I suspect the increasing urgency of replacing laminas-db is going to take up a lot of my time before I get around to working on this important but less-vital goal.

In any case, getting back down to the task at hand, if it would make sense to establish a HoldsController and start splitting some functionality away now, maybe that would offer an excuse to begin some of the necessary refactoring... but if that makes the rest of this project harder, we can just leave it as-is for now and deal with the refactoring later.

Reply:
Introducing a HoldsController now would make sense. I'll just do that and move HoldsAction there as well. Do you have a preference on whether to keep the holds.phtml template in the current location for back-compatibility? I'm tempted to move it as well to make its location standard for the future, but that'd be an incompatible change. I think that keeping the old route is necessary.

Reply:
I don't object to moving holds.phtml as long as we document it in the changelog. Would it make sense to move the existing Holds functionality as a separate PR and then merge that work here, to keep things broken into smaller chunks? If that makes the work harder, though, I'm fine with having it done here -- we'll just want to add a TODO checkbox about the changelog so I don't lose track of it.

Reply:
I almost put it here, but decided it's so much cleaner to do the refactoring separately and posted PR #1961. Now I just need it in to continue here...